src/common.speech/ConversationTranscriberConnectionFactory.ts

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import {
+    ProxyInfo,
+    WebsocketConnection,
+} from "../common.browser/Exports";
+import {
+    IConnection,
+    IStringDictionary
+} from "../common/Exports";
+import {
+    PropertyId
+} from "../sdk/Exports";
+import {
+    ServicePropertiesPropertyName
+} from "../common.speech/Exports";
+import {
+    ConnectionFactoryBase
+} from "./ConnectionFactoryBase";
+import {
+    AuthInfo,
+    RecognizerConfig,
+    WebsocketMessageFormatter
+} from "./Exports";
+import { HeaderNames } from "./HeaderNames";
+import {
+    QueryParameterNames
+} from "./QueryParameterNames";
+
+export class ConversationTranscriberConnectionFactory extends ConnectionFactoryBase {
+    private readonly universalUri: string = "/speech/universal/v2";
+
+    public create(
+        config: RecognizerConfig,
+        authInfo: AuthInfo,
+        connectionId?: string): IConnection {
+
+        let endpoint: string = config.parameters.getProperty(PropertyId.SpeechServiceConnection_Endpoint, undefined);
+        const region: string = config.parameters.getProperty(PropertyId.SpeechServiceConnection_Region, undefined);
+        const hostSuffix: string = ConnectionFactoryBase.getHostSuffix(region);
+        const host: string = config.parameters.getProperty(PropertyId.SpeechServiceConnection_Host, "wss://" + region + ".stt.speech" + hostSuffix);
+        const queryParams: IStringDictionary<string> = {};
+        const endpointId: string = config.parameters.getProperty(PropertyId.SpeechServiceConnection_EndpointId, undefined);
+        const language: string = config.parameters.getProperty(PropertyId.SpeechServiceConnection_RecoLanguage, undefined);
+
+        if (endpointId) {
+            if (!endpoint || endpoint.search(QueryParameterNames.CustomSpeechDeploymentId) === -1) {
+                queryParams[QueryParameterNames.CustomSpeechDeploymentId] = endpointId;
+            }
+        } else if (language) {
+            if (!endpoint || endpoint.search(QueryParameterNames.Language) === -1) {
+                queryParams[QueryParameterNames.Language] = language;
+            }
+        }
+
+        if (config.autoDetectSourceLanguages !== undefined) {
+            queryParams[QueryParameterNames.EnableLanguageId] = "true";
+        }
+
+        this.setV2UrlParams(config, queryParams, endpoint);
```

Comment:
setting these URL parameters is unique to V2?

Reply:
ConversationTranscriber always uses v2, these are the URL params that are not currently sent as json.